tasks/ModelTask.cpp

```diff
@@ -54,44 +58,95 @@ void ModelTask::setupJoints()
 void ModelTask::setupLinks()
 {
     // The robot configuration YAML file must define the exported links.
-    exported_links_list = _exported_links.get();
-    for(std::vector<LinkExport>::iterator it = exported_links_list.begin();
-            it != exported_links_list.end(); ++it)
+    vector<LinkExport> export_conf = _exported_links.get();
+    set<string> port_names;
+    for(vector<LinkExport>::iterator it = export_conf.begin();
+            it != export_conf.end(); ++it)
     {
-        (*it).source_link = checkExportedLinkElements("source_link", (*it).source_link, "world");
-        (*it).target_link = checkExportedLinkElements("target_link", (*it).target_link, "world");
-        (*it).source_frame = checkExportedLinkElements("source_frame", (*it).source_frame, (*it).source_link);
-        (*it).target_frame = checkExportedLinkElements("target_frame", (*it).target_frame, (*it).target_link);
-
-        LinkPtr source_link = model->GetLink( (*it).source_link );
-        if ( !source_link )
+        ExportedLink exported_link;
+
+        exported_link.source_link =
+            checkExportedLinkElements("source_link", it->source_link, "world");
+        exported_link.target_link =
+            checkExportedLinkElements("target_link", it->target_link, "world");
+        exported_link.source_frame =
+            checkExportedLinkElements("source_frame", it->source_frame, it->source_link);
+        exported_link.target_frame =
+            checkExportedLinkElements("target_frame", it->target_frame, it->target_link);
+        exported_link.source_link_ptr = model->GetLink( it->source_link );
+        exported_link.target_link_ptr = model->GetLink( it->target_link );
+
+        if (it->source_link != "world" && !exported_link.source_link_ptr)
+        {
+            gzmsg << "ModelTask: cannot find link " << it->source_link << " in model" << endl;
+            gzmsg << "Exiting simulation." << endl;
```

Comment:
We should call gzthrow() here because these errors are related to the configuration file right !?